python-regression/tests/features/machine6/6_local_snapshots_tests.feature

```diff
@@ -84,4 +84,50 @@ Feature: Test Bootstrapping With LS
       |keys                       |values                   |type             |
       |tails                      |LS_PRUNED_TRANSACTIONS   |staticValue      |
 
-    Then the response for "checkConsistency" should return null
\ No newline at end of file
+    Then the response for "checkConsistency" should return null
+
+
+  Scenario: Check unconfirmed transaction is spent from
+    Issues a value transaction that will be unconfirmed, and check that the address was spent from.
+
+    Given a transaction is generated and attached on "nodeE" with:
+      |keys                       |values                   |type           |
+      |address                    |TEST_ADDRESS             |staticValue    |
+      |value                      |10                       |int            |
+      |seed                       |UNCONFIRMED_TEST_SEED    |staticValue    |
+
+    When "wereAddressesSpentFrom" is called on "nodeE" with:
+      |keys                       |values                   |type             |
+      |addresses                  |UNCONFIRMED_TEST_ADDRESS |staticValue      |
+
+    Then the response for "wereAddressesSpentFrom" should return with:
+      |keys                       |values                   |type             |
+      |addresses                  |True                     |boolList         |
+
+
+
+  Scenario: Check addresses spent from after pruning
+    Ensures that a node with a spent address registers that the address is spent from both before and after the
+    transaction has been pruned from the DB.
+
+    # Check that addresses were spent from before pruning
+    Given "wereAddressesSpentFrom" is called on "nodeE" with:
+      |keys                       |values                   |type             |
+      |addresses                  |LS_SPENT_ADDRESSES       |staticValue      |
+
+    Then the response for "wereAddressesSpentFrom" should return with:
+      |keys                       |values                   |type             |
+      |addresses                  |True                     |boolList         |
+
+    When the next 100 milestones are issued
+
+    # Check that addresses were spent after transaction have been pruned
+    And "wereAddressesSpentFrom" is called on "nodeE" with:
+      |keys                       |values                   |type             |
+      |addresses                  |LS_SPENT_ADDRESSES       |staticValue      |
+
+    Then the response for "wereAddressesSpentFrom" should return with:
+      |keys                       |values                   |type             |
+      |addresses                  |True                     |boolList         |
+
```

Comment:
Change Request:
Can you try to fetch the tx from the db and fail if you succeed?
This is to raise confidence that pruning did occur

Reply:
@DyrellC 
I saw you pushed but didn't see you address this

If you think this shouldn't be done write a comment (I want this though, so I'll be pushy)

Reply:
I did address this. I added extra steps to the gherkin file just below

Reply:
@galrogo